### ISSUE TITLE
Overriding issues 

### ISSUE THREAD

**Issue by Scnck** [wenzhixin/bootstrap-table#6558]
### Description

I'm no longer able to restore data changes back to original data source by overriding function("initData").  What I'm I missing?
```
<!doctype html>
<html lang="en">
  <head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    <!-- Bootstrap CSS -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65" crossorigin="anonymous">
	<link rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.21.2/dist/bootstrap-table.min.css">
	
    <title>TEST!</title>
  </head>
  <body>
	<div id="toolbar">
	  <button id="getData" type="submit" class="btn btn-primary">getData</button>
	</div>
	<table
		id="table"
		data-toolbar="#toolbar">
	  <thead>
		<tr>
		  <th data-field="checked" data-checkbox="true"></th>
		  <th data-field="id">ID</th>
		  <th data-field="name">Item Name</th>
		  <th data-field="price">Item Price</th>
		</tr>
	  </thead>
	</table>

    <!-- Option 1: jQuery and Bootstrap Bundle (includes Popper) -->
    <script src="https://code.jquery.com/jquery-3.6.2.min.js" integrity="sha256-2krYZKh//PcchRtd+H+VyyQoZ/e3EcrkxhM8ycwASPA=" crossorigin="anonymous"></script>
	<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-kenU1KFdBIe4zVF0s0G1M5b4hcpxyD9F7jL+jjXkk+Q2h455rYXK/7HAuoJl+0I4" crossorigin="anonymous"></script>
	<script src="https://unpkg.com/bootstrap-table@1.21.2/dist/bootstrap-table.min.js"></script>
	<script>
		(function($) {

			"use strict";

			let BootstrapTable = $.fn.bootstrapTable.Constructor,
				_initData = BootstrapTable.prototype.initData;

			BootstrapTable.prototype.initData = function (data, type) {
				if (type === "append") {
					this.options.data = this.options.data.concat(data)
				} else if (type === "prepend") {
					this.options.data = [].concat(data).concat(this.options.data)
				} else {
					data = data || this.options.data;
					this.options.data = Array.isArray(data) ? data : data[this.options.dataField]
				}

				this.data = [...this.options.data]

				if (this.options.sortReset) {
					this.unsortedData = [...this.data]
				}

				if (this.options.sidePagination === "server") {
					return
				}
				this.initSort()
				
				console.log("_initData");
			};

		})(jQuery);
	</script>
	<script>
		let $table = $('#table')
		let $getData = $('#getData');
		let data = [
		  {
			'checked': false,
			'id': 0,
			'name': 'Item 0',
			'price': '$0'
		  },
		  {
			'checked': false,
			'id': 1,
			'name': 'Item 1',
			'price': '$1'
		  },
		  {
			'checked': false,
			'id': 2,
			'name': 'Item 2',
			'price': '$2'
		  },
		  {
			'checked': false,
			'id': 3,
			'name': 'Item 3',
			'price': '$3'
		  },
		  {
			'checked': false,
			'id': 4,
			'name': 'Item 4',
			'price': '$4'
		  },
		  {
			'checked': true,
			'id': 5,
			'name': 'Item 5',
			'price': '$5'
		  }
		]

		$(function() {
			$getData.click(function () {
				let lData = $table.bootstrapTable('getData');
				alert(JSON.stringify(lData))
				alert(JSON.stringify(data))
			})

			$table.bootstrapTable({data: data})
		});
	</script>
  </body>
</html>
```

**Comment by Scnck** [comment]
the Data source should be referenced by ref vs copy, based on my overriding function.  This is the way the plugin was initially designed and has since been changed.. but my overloading function has worked until this latest release but I'm not seeing what change is preventing my logic from still working.  if you change the reference to bootstrap-table@1.21.1 my logic works and when the data is checked within the tables it changes the original data source("data")

let data = [....];
https://javascript.info/object-copy

**Comment by wenzhixin** [comment]
@Scnck Please help check if this PR: https://github.com/wenzhixin/bootstrap-table/pull/6560 solves your problem.

**Comment by Scnck** [comment]
why I'm I not allowed to overwrite the $.fn.bootstrapTable.utils.extend function?

var getSearchInput  = $.fn.bootstrapTable.utils.getSearchInput;
var sprintf  = $.fn.bootstrapTable.utils.sprintf;
var extend  = $.fn.bootstrapTable.utils.extend;

console.log(getSearchInput);
console.log(sprintf);
console.log(extend);

**Comment by wenzhixin** [comment]
You need to use the develop branch: https://live.bootstrap-table.com/code/wenzhixin/13763.

**Comment by wenzhixin** [comment]
The changes here have not yet been released.

**Comment by UtechtDustin** [comment]
Should be fixed by #6560, right ?
If not, please let us know!
